pkg/components/pluggable/loader.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2021 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pluggable
+
+import (
+	"os"
+
+	componentsV1alpha1 "github.com/dapr/dapr/pkg/apis/components/v1alpha1"
+	"github.com/dapr/dapr/pkg/components"
+)
+
+// mapComponents maps from pluggablecomponent manifest to pluggable component obj.
+func mapComponents(comps []componentsV1alpha1.PluggableComponent) []components.Pluggable {
+	pluggableComponents := make([]components.Pluggable, len(comps))
+	for idx, component := range comps {
+		pluggableComponents[idx] = components.Pluggable{
+			Name:    component.GetObjectMeta().GetName(),
+			Type:    components.PluggableType(component.Spec.Type),
+			Version: component.Spec.Version,
+		}
+	}
+	return pluggableComponents
+}
+
+// newFromPath creates a disk manifest loader from the given path.
+func newFromPath(pluggableComponentsPath string) components.ManifestLoader[componentsV1alpha1.PluggableComponent] {
+	return components.NewDiskManifestLoader(pluggableComponentsPath, func() componentsV1alpha1.PluggableComponent {
+		var comp componentsV1alpha1.PluggableComponent
+		comp.Spec = componentsV1alpha1.PluggableComponentSpec{}
+		return comp
+	})
+}
+
+// LoadFromDisk loads PluggableComponents from the given path.
+func LoadFromDisk(pluggableComponentsPath string) ([]components.Pluggable, error) {
+	comp, err := newFromPath(pluggableComponentsPath).Load()
+	if os.IsNotExist(err) {
+		return make([]components.Pluggable, 0), nil
+	}
+	return mapComponents(comp), err
+}
+
+// TODO Load from K8S
```

Comment:
Should we have an issue to track this TODO?

Reply:
Dont know if we need an issue for that, I think the overall progress of pluggable components must include this, hence the current issue should be sufficient, what do you think?

Reply:
I don't know what is the practice for this project. If this is being addressed in a follow up PR, I don't see the reason for it. If this is just a "good to have" feature then, again, an issue would not really be required. For all other cases I would say it is worth having an issue open to keep track of this -- no one is keeping track of the number of open TODOs, right? :-P

Reply:
tbh I agree with you that we should have "small" issues instead of a big one that should be closed when everything is ready - especially because being ready is too relative.

One thing that would become confuse is when the big issue should be closed then, perhaps it shouldn't be treat as an issue but a milestone/project/something bigger instead? 

anyways I'll create these issues shortly after having this PR closed when we have more clear vision of what we need in the near future, how about that?